api/tests/src/api/opportunities_v0_1/test_opportunity_route.py

```diff
@@ -987,6 +1093,211 @@ def test_opportunity_search_filters_200(
+            (
+                get_search_request(
+                    page_size=25,
+                    query="humankind",
+                    applicant_type_one_of=[
+                        ApplicantType.STATE_GOVERNMENTS,
+                        ApplicantType.FEDERALLY_RECOGNIZED_NATIVE_AMERICAN_TRIBAL_GOVERNMENTS,
+                    ],
+                ),
+                [
+                    Scenario.POSTED_ALL_ENUM_VALUES,
+                    Scenario.POSTED_NON_DEFAULT_AGENCY_WITH_APP_TYPES,
+                ],
+            ),
+            # Like the previous one, but the query is more specific and only gets one of the scenarios
+            (
+                get_search_request(
+                    page_size=25,
+                    query="very few enum known to humankind",
+                    applicant_type_one_of=[
+                        ApplicantType.STATE_GOVERNMENTS,
+                        ApplicantType.FEDERALLY_RECOGNIZED_NATIVE_AMERICAN_TRIBAL_GOVERNMENTS,
+                    ],
+                ),
+                [Scenario.POSTED_NON_DEFAULT_AGENCY_WITH_APP_TYPES],
+            ),
+            # Agency filtered by one_of, query hits something in the summary description
+            (
+                get_search_request(
+                    page_size=25,
+                    query="i am a description",
+                    agency_one_of=["DIFFERENT-XYZ", "DIFFERENT-abc"],
+                ),
+                [Scenario.CLOSED_NON_DEFAULT_AGENCY_WITH_FUNDING_CATEGORIES],
+            ),
+            ### A few scenarios that don't return any results because filters/query make it too specific
+            (
+                get_search_request(
+                    page_size=25,
+                    query="humankind",
+                    opportunity_status_one_of=[
+                        OpportunityStatus.FORECASTED,
+                        OpportunityStatus.CLOSED,
+                        OpportunityStatus.ARCHIVED,
+                    ],
+                ),
+                [],
+            ),
+            (
+                get_search_request(
+                    page_size=25,
+                    query="different",
+                    funding_instrument_one_of=[FundingInstrument.GRANT],
+                ),
+                [],
+            ),
+            (
+                get_search_request(
+                    page_size=25,
+                    query="words that don't hit anything",
+                    applicant_type_one_of=[
+                        ApplicantType.STATE_GOVERNMENTS,
+                        ApplicantType.FEDERALLY_RECOGNIZED_NATIVE_AMERICAN_TRIBAL_GOVERNMENTS,
+                    ],
+                ),
+                [],
+            ),
+        ],
+        ids=search_scenario_id_fnc,
+    )
+    def test_opportunity_query_and_filter_200(
+        self, client, api_auth_token, search_request, expected_scenarios, setup_scenarios
```

Comment:
is the `setup_scenarios` fixture being used?

Reply:
It is used by all the tests in this class. It only gets called once for the set of tests for the class, but is required by all of them as it is where we setup the data we actually query against.